.github/workflows/rust.yml

```
@@ -42,6 +42,9 @@ jobs:
     - name: Clippy check
       run: cargo clippy --all-targets
 
+    - name: Clippy check with all features
+      run: cargo clippy --all-targets --all-features
```

```suggestion
    - name: Clippy check with opensearch feature
      run: cargo clippy --all-targets --features opensearch
```
The `default` and the `opensearch` features are exclusive, so it is better to run steps for them separately, and name them accordingly. There should be also additional steps for `cargo fmt` and `cargo test`

---

The commit and PR description should be also changed as this patch should introduce only CI tests for `opensearch`

---

The commit and PR description should be also changed as this patch should introduce only CI tests for `opensearch`